internal/services/maintenance/maintenance_configuration_resource.go

```diff
@@ -293,14 +293,73 @@ func resourceArmMaintenanceConfigurationCreateUpdate(d *pluginsdk.ResourceData,
 	}
 
 	if _, err := client.CreateOrUpdate(ctx, id, configuration); err != nil {
-		return fmt.Errorf("creating/updating %s: %+v", id, err)
+		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
-	return resourceArmMaintenanceConfigurationRead(d, meta)
+	return resourceMaintenanceConfigurationRead(d, meta)
 }
 
-func resourceArmMaintenanceConfigurationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+func resourceMaintenanceConfigurationUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.ConfigurationsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```

Comment:
_minor_ but this can become:

```suggestion
	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
```

(same in the create)